jax/tree_util.py

```diff
@@ -172,28 +171,66 @@ def __ne__(self, other):
 
 
 class PyLeaf(object):
+  include_subclasses = True
   def __repr__(self):
     return '*'
 
 leaf = PyLeaf()
 
 def dict_to_iterable(xs):
-  keys = tuple(sorted(xs.keys()))
-  return tuple(map(xs.get, keys)), keys
+  keys, values = zip(*sorted(xs.items()))
+  return values, keys
 
 class NodeType(object):
-  def __init__(self, name, to_iterable, from_iterable):
+  def __init__(self, name, to_iterable, from_iterable, include_subclasses):
     self.name = name
     self.to_iterable = to_iterable
     self.from_iterable = from_iterable
+    self.include_subclasses = include_subclasses
 
 node_types = {}
 
-def register_pytree_node(py_type, to_iterable, from_iterable):
+def get_node_type(py_type):
+  for cls in py_type.__mro__:
+    node_type = node_types.get(cls)
+    if node_type is not None and (
+        cls is py_type or node_type.include_subclasses):
+      return node_type
+  return leaf
+
+def register_pytree_node(py_type, to_iterable, from_iterable,
+                         include_subclasses=True):
```

Comment:
Can we not have a default value here, or else leave the default to `False` so that previous behavior is maintained? In general I think default values can be bad and it's better to force the call site to be explicit.